### PR TITLE
Allow Signing only in Mgmt Pipeline

### PR DESCRIPTION
**Pull request by chidozieononiwu**
-Modify mgmt release job to allow for only signing.
-Leaving `ShouldPublish` unchecked will result in signed packages.

### CODE REVIEW

File: eng/pipelines/templates/jobs/mgmt-release.yml [hunk 1]
```diff
@@ -25,52 +28,55 @@ jobs:
...
+              -packageRepository 'Nuget'
+              -releaseSha '$(Build.SourceVersion)'
+              -repoOwner 'Azure'
+              -repoName 'azure-sdk-for-net'
+            pwsh: true
+          env:
+            GH_TOKEN: $(azuresdk-github-pat)
+          condition: and(succeeded(), ne(variables['SkipCreateTagAndGitRelease'], 'true'))
 
       - template: pipelines/steps/net-signing.yml@azure-sdk-build-tools
```
Comment (weshaggard) [inline eng/pipelines/templates/jobs/mgmt-release.yml:48]: Let's move this signing steps to be the first thing in this job and then we can condition all the other steps (except artifact publishing) with the condition you are using. That way we don't have to duplicate that condition so much.

File: eng/pipelines/templates/jobs/mgmt-release.yml [hunk 2]
```diff
@@ -25,52 +28,55 @@ jobs:
...
-          publishFeedCredentials: Nuget.org
-        condition: and(succeeded(), eq(variables['Build.Repository.Name'], 'Azure/azure-sdk-for-net'))
+        - task: NuGetCommand@2
+          displayName: 'Publish to Nuget'
+          inputs:
+            command: push
+            packagesToPush: '$(PIPELINE.WORKSPACE)/packages/**/*.nupkg;!$(PIPELINE.WORKSPACE)/packages/**/*.symbols.nupkg'
+            nuGetFeedType: external
+            publishFeedCredentials: Nuget.org
+          condition: succeeded()
```
Comment (weshaggard) [inline eng/pipelines/templates/jobs/mgmt-release.yml:79]: Condition isn't necessary.

File: eng/pipelines/templates/jobs/mgmt-release.yml [hunk 3]
```diff
@@ -25,52 +28,55 @@ jobs:
...
-        displayName: 'Use NuGet ${{ parameters.NugetVersion }}'
-        inputs:
-          versionSpec: ${{ parameters.NugetVersion }}
-        condition: and(succeeded(), eq(variables['Build.Repository.Name'], 'Azure/azure-sdk-for-net'))
+      - ${{ if and(eq(parameters.ShouldPublish, 'true'), eq(variables['Build.Repository.Name'], 'Azure/azure-sdk-for-net')) }}:
+        - task: NuGetToolInstaller@1
+          displayName: 'Use NuGet ${{ parameters.NugetVersion }}'
+          inputs:
+            versionSpec: ${{ parameters.NugetVersion }}
+          condition: succeeded()
```
Comment (weshaggard) [inline eng/pipelines/templates/jobs/mgmt-release.yml:70]: condition isn't necessary

File: eng/pipelines/templates/jobs/mgmt-release.yml [hunk 4]
```diff
@@ -25,52 +28,55 @@ jobs:
...
+        - task: MSBuild@1
+          displayName: 'Upload Symbols'
+          inputs:
+            solution: '${{ parameters.BuildToolsPath }}/tools/symboltool/SymbolUploader.proj'
+            msbuildArguments: >
+              /p:PackagesPath=$(PIPELINE.WORKSPACE)/packages
+              /p:MSPublicSymbolsPAT=$(azuresdk-microsoftpublicsymbols-devops-pat)
+              /p:MSSymbolsPAT=$(azuresdk-microsoft-devops-pat)
+              /p:AzureSDKSymbolsPAT=$(azuresdk-azure-sdk-devops-pat)
+          condition: succeeded()
```
Comment (weshaggard) [inline eng/pipelines/templates/jobs/mgmt-release.yml:63]: condition isn't necessary.

File: eng/pipelines/templates/jobs/mgmt-release.yml (hunk 1 above)
Comment (weshaggard) [reply eng/pipelines/templates/jobs/mgmt-release.yml:48]: Looks like you correctly grouped the bottom set of steps but I still think we can group the tagging with the below steps by moving the signing to be the first step.

Comment (chidozieononiwu) [reply eng/pipelines/templates/jobs/mgmt-release.yml]: Sorry I failed to push the latest changes.

**Review by weshaggard — APPROVED** [review]

**Pull request merged**